securedrop/tests/functional/test_source.py

```diff
@@ -17,3 +18,16 @@ def test_lookup_codename_hint(self):
         self._source_chooses_to_login()
         self._source_proceeds_to_login()
         self._source_sees_no_codename()
+
+
+class TestDownloadKey(
+        functional_test.FunctionalTest,
+        journalist_navigation_steps.JournalistNavigationStepsMixin):
+
+    def test_journalist_key_from_source_interface(self):
+        data = self.return_downloaded_content(self.source_location +
+                                              "/journalist-key", None)
+
+        if six.PY3:
+            data = data.decode('utf-8')
+        assert "BEGIN PGP PUBLIC KEY BLOCK" in data
```

Comment:
the corresponding unit test in `tests/test_source.py::test_lookup` did not fail because the assert causing the failure ([`assert isinstance(data, bytes), 'applications must write bytes'`](https://github.com/pallets/werkzeug/blob/master/src/werkzeug/serving.py#L274) ) from `werkzeug/serving.py` does not occur in test since the dev server is not running: this underscores the importance of having full _functional test_ coverage of all endpoints